loopy/kernel/instruction.py

```diff
@@ -172,31 +171,9 @@ def __init__(self, id, depends_on, depends_on_is_final,
             no_sync_with,
             within_inames_is_final, within_inames,
             priority,
-            boostable, boostable_into, predicates, tags,
-            insn_deps=None, insn_deps_is_final=None,
+            predicates, tags,
             forced_iname_deps=None, forced_iname_deps_is_final=None):
 
-        # {{{ backwards compatibility goop
-
-        if depends_on is not None and insn_deps is not None:
-            raise LoopyError("may not specify both insn_deps and depends_on")
-        elif insn_deps is not None:
-            warn("insn_deps is deprecated, use depends_on",
-                    DeprecationWarning, stacklevel=2)
-
-            depends_on = insn_deps
-            depends_on_is_final = insn_deps_is_final
-
-        if forced_iname_deps is not None and within_inames is not None:
-            raise LoopyError("may not specify both forced_iname_deps "
-                    "and within_inames")
-        elif forced_iname_deps is not None:
-            warn("forced_iname_deps is deprecated, use within_inames",
-                    DeprecationWarning, stacklevel=2)
-
-            within_inames = forced_iname_deps
-            within_inames_is_final = forced_iname_deps_is_final
-
```

Comment:
Should also remove the corresponding args if this is removed, to make sure the args aren't silently ignored. The warning didn't have a deadline, but it's super-old, so let's kill it anyway.

Reply:
Done in 96af643e.